File: Arrays/Java/Minimum_Difference_among_K.java
class Solution
{
    long minDiff(long a[] ,int N,int K)
    {
    
        Arrays.sort(a);
        long ans = Integer.MAX_VALUE;
        for (int i=0; i<=N-K; i++) {
            long min = a[i];
            long max = a[i+K-1];
            long d = max-min;
            if (ans > d) {
                ans = d;
            }
        }
        return ans;
    }
}

File: Solutions/Python/Minimum_Difference_among_K.py
class Solution:
    def minDiff(self,a,N,K):
        a.sort()
        ans = a[-1]
        for i in range(0, N-K+1):
            mini = a[i]
            maxi = a[i+K-1]
            d = maxi - mini
            if ans > d:
                ans = d
        return ans
